### ISSUE TITLE
Question regarding usage of gradient_penalty in Soundstream

### ISSUE THREAD

**Issue by gdoras** [lucidrains/audiolm-pytorch#290]
Hello, in the original WGAN with gradient penalty [paper](https://arxiv.org/abs/1704.00028) , the penalty enforces the norm of gradient of the output of the discriminator wrt its input (i.e. the reconstruction from the decoder) to be equal to 1:

$$
\mathcal{L}_{GP} = \lambda .  ( \lVert \nabla_{\tilde{x}} D(\tilde{x}) \rVert_2 - 1)^2 
$$

In the Soundstream [code](https://github.com/lucidrains/audiolm-pytorch/blob/main/audiolm_pytorch/soundstream.py#L884), the gradient computed is the gradient from the loss wrt to the inputs:

```python
if apply_grad_penalty:
     stft_grad_penalty = gradient_penalty(real, stft_discr_loss) + gradient_penalty(fake, stft_discr_loss)
```

(same for the other losses).

Is this really correct ? Or should not it be:

```python
if apply_grad_penalty:
     stft_grad_penalty = gradient_penalty(real, stft_real_logits, center=1.) + gradient_penalty(fake, stft_fake_logits, center=1.)
```
